File: ruby/lib/sorpresa_porjugador.rb
module Civitas
    class SorpresaPorJugador < Sorpresa
        public_class_method :new
        def initialize(texto, valor)
            super(texto)
            @valor=valor
        end

        def aplicarAJugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                s1= SorpresaPagarCobrar.new("TODOS", @valor*-1)
                for i in 0...todos.length do
                    if i!=actual
                        s1.aplicarAJugador(i, todos)
                    end
                end

                sorpresa= SorpresaPagarCobrar.new("ACTUAL", @valor*(todos.length-1))
                sorpresa.aplicarAJugador(actual, todos)
            end
        end
    end
end

File: Civitas_gui/src/civitas/SorpresaPorJugador.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorJugador extends Sorpresa {
    private int valor;

    SorpresaPorJugador(String texto, int valor) {
        super(texto);
        this.valor = valor;
    }
    

    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual,todos);
            Sorpresa s1 = new SorpresaPagarCobrar("Quita", valor * -1);
            for (int i = 0; i < todos.size(); i++) {
                if (i != actual)
                    s1.aplicarAJugador(i, todos);
            }

            Sorpresa s2 = new SorpresaPagarCobrar("Pon", valor * (todos.size() - 1));
            s2.aplicarAJugador(actual, todos);
        }
    }
}